modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/manager/v1_0/DefaultObjectEntryManagerImpl.java

```diff
@@ -764,13 +763,19 @@ private void _checkObjectEntryObjectDefinitionId(
 	}
 
 	private ServiceContext _createServiceContext(
-		Map<String, Object> properties, long userId) {
+		ObjectEntry objectEntry, long userId) {
 
 		ServiceContext serviceContext = new ServiceContext();
 
 		serviceContext.setAddGroupPermissions(true);
 		serviceContext.setAddGuestPermissions(true);
 
+		if (Validator.isNotNull(objectEntry.getKeywords())) {
+			serviceContext.setAssetTagNames(objectEntry.getKeywords());
```

Comment:
Could this be overwritten? 12 lines below we have the following code:
```
if (properties.get("tagNames") != null) {
			serviceContext.setAssetTagNames(
				ArrayUtil.toStringArray(
					(List<String>)properties.get("tagNames")));
		}
```
This means that if the Object comes with a property called `tagNames`, the keywords set here will be overwritten.

That makes me wonder if keywords have some support and it used through the portal. I found [this code](https://github.com/liferay/liferay-portal/blob/8b6b423d4c583c4897fc1ac2d4c8de31eafe1cfb/modules/apps/object/object-web/src/main/resources/META-INF/resources/object_entries/object_entry/form.jsp#L166-L169) to create an object entry with categorization. Should we change this to use the keywords property instead of `tagNames`

Reply:
Yeah I saw it too. But I didn't want to break backwards compatibility. We can send a follow up PR replacing that code once the feature flag is removed. I created https://issues.liferay.com/browse/LPS-178317 for this.

Thanks!!